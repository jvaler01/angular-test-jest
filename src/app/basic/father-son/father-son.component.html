<h1>Son</h1>
<hr>
<code class="mt-2">{{ client | json }}</code>
<hr>
<button data-test="btn-update" *ngIf="client" class="btn btn-outline-primary"(click)="onChange(2)">
    Change Id
</button>
<button data-test="btn-delete" *ngIf="client" class="btn btn-outline-danger ml-2" (click)="onDelete()">
    Remove client
</button>