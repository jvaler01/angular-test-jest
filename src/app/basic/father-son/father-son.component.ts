import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { Client } from '../interfaces';

@Component({
  selector: 'app-father-son',
  templateUrl: './father-son.component.html',
  styleUrls: ['./father-son.component.css']
})
export class FatherSonComponent implements OnInit {

  @Input()
  client?: Client;
  @Output() 
  onClientUpdated = new EventEmitter<Client>();
  @Output() 
  onClientDeleted = new EventEmitter();

  constructor() { }

  ngOnInit(): void {
  }

  onDelete(){
    this.client = undefined;
    this.onClientDeleted.emit();
  }

  onChange( id: number){
    if( !this.client ) return;
    this.client = {
      ...this.client,
      id
    }
    this.onClientUpdated.emit({ ...this.client });
  }
}
