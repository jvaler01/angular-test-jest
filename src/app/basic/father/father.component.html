<h1>Father</h1>
<button class="btn btn-primary" (click)="onSetClient('test')">
    Load Client
</button>
<hr>
<code class="mt-2">{{ client | json }}</code>
<hr>
<app-father-son 
    [client]="client"
    (onClientDeleted)="client = undefined"
    (onClientUpdated)="client = $event"
>
</app-father-son>