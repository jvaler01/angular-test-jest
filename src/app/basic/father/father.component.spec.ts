import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { FatherSonComponent } from '../father-son/father-son.component';
import { FatherComponent } from './father.component';

describe('FatherComponent', () => {
  let component: FatherComponent;
  let fixture: ComponentFixture<FatherComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ FatherComponent, FatherSonComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FatherComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('check snapshoot', () => {
    expect( compiled ).toMatchSnapshot();
  });

  it('check set client', () => {
    component.onSetClient('Test');
    fixture.detectChanges();

    const element = compiled.querySelector('.mt-2');

    expect( element?.textContent ).toContain('"name"');
    expect( element?.textContent ).toContain('"Test"');
  });

  it('check onDelete event', () => {
    component.client = { id: 1, name: 'Test'};
    fixture.detectChanges();

    const sonComponentDebug = fixture.debugElement.query( By.directive(FatherSonComponent) );
    const sonComponent = sonComponentDebug.componentInstance;

    sonComponent.onClientDeleted.emit();
    expect( component.client ).toBe(undefined);
  });

  it('check onUpdate event', () => {
    component.client = { id: 1, name: 'Test'};
    fixture.detectChanges();

    const sonComponentDebug = fixture.debugElement.query( By.directive(FatherSonComponent) );
    const sonComponent = sonComponentDebug.componentInstance;

    sonComponent.onClientUpdated.emit({ id: 5, name: 'Test' });
    //To Be for primitive
    //To Be Equal pointer
    //To Equal Element by Element
    expect( component.client ).toEqual({ id: 5, name: 'Test' });
  });
});
