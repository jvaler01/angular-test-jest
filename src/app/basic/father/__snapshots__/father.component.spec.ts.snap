// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatherComponent check snapshoot 1`] = `
<div
  id="root1"
>
  <h1>
    Father
  </h1>
  <button
    class="btn btn-primary"
  >
     Load Client

  </button>
  <hr />
  <code
    class="mt-2"
  >
    
  </code>
  <hr />
  <app-father-son>
    <h1>
      Son
    </h1>
    <hr />
    <code
      class="mt-2"
    >
      
    </code>
    <hr />
    
    
  </app-father-son>
</div>
`;
